src/lib.rs

```diff
@@ -12,22 +13,53 @@ fn markdown_html_finder(_py: Python, m: &PyModule) -> PyResult<()> {
     /// Find positions of html nodes in a markdown file
     ///
     /// Returns a list of tuples of start and end positions
-    fn find_html_positions_py(markdown: &str) -> PyResult<Vec<(usize, usize)>> {
-        Ok(find_html_positions(markdown))
+    ///
+    /// Raises ValueError when passed carriage returns. You must strip carriage
+    /// returns before calling this function.
+    fn find_html_positions_py(markdown: &'static str) -> PyResult<Vec<(usize, usize)>> {
+        find_html_positions(markdown).map_err(|msg| exceptions::ValueError::py_err(msg))
     }
 
     Ok(())
 }
 
-fn find_html_positions(markdown: &str) -> Vec<(usize, usize)> {
-    Parser::new_ext(markdown, Options::empty())
+fn join_adjacent_spans(spans: Vec<(usize, usize)>) -> Vec<(usize, usize)> {
+    let mut new_spans = Vec::new();
+    for span in spans.iter() {
+        let (span_start, span_end) = span;
+        if let Some(prev_span) = new_spans.last_mut() {
+            let (prev_start, prev_end) = prev_span;
+            if prev_end == span_start {
```

Comment:
```suggestion
            if prev_end >= span_start {
```

Reply:
This would never happen in this case, but I'll add a test anyway.